myjsonrpc.pm

```diff
@@ -48,36 +63,47 @@ our $sockets;
 sub read_json {
     my ($socket, $cmd_token) = @_;
 
-    my $cjx = Cpanel::JSON::XS->new;
 
     my $fd = fileno($socket);
+    if ($DEBUG_JSON) {
+        bmwqemu::diag("($$) read_json($fd)");
+    }
+    my $buffer = '';
     if (exists $sockets->{$fd}) {
         # start with the trailing text from previous call
-        $cjx->incr_parse($sockets->{$fd});
-        delete $sockets->{$fd};
+        $buffer = delete $sockets->{$fd};
     }
 
     my $s = IO::Select->new();
     $s->add($socket);
 
-    my $hash;
+    my $multi = wantarray ? 1 : 0;
```

Comment:
I also tend to avoid context.
An alternative would be another parameter to the sub.
It already has a parameter (which is not used a single time in our code AFAICS).

How would you do it?
```
read_json($fh, undef, 1) # multi
read_json($fh, undef, 0) # single
```

Reply:
The 2nd parameter is actually used within `driver.pm`.